src/iotjs_binding.c

```diff
@@ -85,6 +86,42 @@ double iotjs_jval_as_number(jerry_value_t jval) {
 }
 
 
+bool iotjs_jval_get_data(jerry_value_t jval, iotjs_string_t* getbuffer) {
+  iotjs_bufferwrap_t* buffer_wrap = NULL;
+  if(jerry_value_is_string(jval)) {
+    jerry_size_t size = jerry_get_string_size(jval);
+
+    if (size == 0) {
+      *getbuffer = iotjs_string_create();
+      return false;
+    }
+
+    char* buffer = iotjs_buffer_allocate(size + 1);
+    jerry_char_t* jerry_buffer = (jerry_char_t*)(buffer);
```

Comment:
There is no need for the extra variable. The cast can be done here the `buffer` is passed.